addons/website_sale/tests/test_website_sale_image.py

```diff
@@ -229,3 +229,21 @@ def test_01_admin_shop_zoom_tour(self):
         self.assertEqual(images[1].image_1920, red_image)
         self.assertEqual(images[2].image_1920, image_gif)
         self.assertEqual(images[3].image_1920, image_svg)
+
+        # CASE: When uploading a product variant image, we don't want the default_product_tmpl_id from the context to be applied
```

Comment:
I would also test the other case: creating a `product.image` with the additional context and no product_variant_id, to make sure the context is kept in that case (then asserting the image is assigned to the template and not the variant).